site/src/pages/WorkspacePage/permissions.ts

```diff
@@ -0,0 +1,39 @@
+import { Workspace, Template } from "api/typesGenerated";
+
+export const workspaceChecks = (workspace: Workspace, template: Template) =>
+  ({
+    readWorkspace: {
+      object: {
+        resource_type: "workspace",
+        resource_id: workspace.id,
+        owner_id: workspace.owner_id,
+      },
+      action: "read",
+    },
+    updateWorkspace: {
+      object: {
+        resource_type: "workspace",
+        resource_id: workspace.id,
+        owner_id: workspace.owner_id,
+      },
+      action: "update",
+    },
+    updateTemplate: {
+      object: {
+        resource_type: "template",
+        resource_id: template.id,
+      },
+      action: "update",
+    },
+    viewDeploymentValues: {
+      object: {
+        resource_type: "deployment_config",
+      },
+      action: "read",
+    },
+  }) as const;
```

Comment:
Do you think it would help to have a separate type defined that lists what each value of the workspace checks object is allowed to be?

It looks like the type is based on the `AuthorizationCheck` type from `typesGenerated.ts`?